Document Clans, Member Verification, and Join Requests (#224)
discord-userdoccers/discord-userdoccers: pages/resources/clan.mdx

# Clan Resource

Clans are an extension of guilds that allow users to apply to join a clan's guild and use a clan's tag globally.

### Clan Object

###### Clan Structure

| Field                 | Type             | Description                                                                 |
| --------------------- | ---------------- | --------------------------------------------------------------------------- |
| id                    | snowflake        | The ID of the guild                                                         |
| name                  | string           | The name of the guild                                                       |

> | tag                   | string           | The tag of the clan (2-4 letters)                                           |

Reviewer: ```suggestion
| tag                   | string           | The tag of the clan (2-4 characters)                                        |
```

| icon_hash             | ?string          | The guild's [icon hash](/reference#cdn-formatting)                          |

> | playstyle             | integer          | Number representing a [playstyle](#playstyle)                               |

Reviewer: ```suggestion
| playstyle             | integer          | The [playstyle](#playstyle-type) of the clan
```

| member_count          | integer          | Approximate count of total members in the guild                             |

> | description           | string           | Description of the clan, this can be different than the guild's description |

Reviewer: ```suggestion
| description           | string           | Description of the clan; this can be different than the guild's description |
```

| search_terms          | array[string]    | A list of 3 or more interests/topics/traits                                 |

> | game_ids              | array[snowflake] | Array of application IDs                                                    |

Reviewer: ```suggestion
| game_ids              | array[snowflake] | The IDs of the applications representing the games the clan plays
```

> | badge                 | integer          | Number representing a [badge kind](#badge-kind)                             |

Reviewer: ```suggestion
| badge                 | integer          | The [badge shown on the clan's tag](#badge-type)                             |
```

| badge_color_primary   | string           | The primary color of the badge as a hexadecimal color string                |
| badge_color_secondary | string           | The secondary color of the badge as a hexadecimal color string              |

> | banner                | integer          | Number representing a [banner style](#banner-style)                         |

Reviewer: ```suggestion
| banner                | integer          | The [banner representing the clan](#banner-style)                         |
```

| brand_color_primary   | string           | The primary color of the banner and some text as a hexadecimal color string |
| brand_color_secondary | string           | The secondary color of the banner as a hexadecimal color string             |
| wildcard_descriptors  | array[string]    | A list of 3 words to describe the clan                                      |
| badge_hash            | string           | The clan's [badge hash](/reference#cdn-formatting)                          |
| banner_hash           | string           | The clan's [banner hash](/reference#cdn-formatting)                         |

> ###### Playstyle

Reviewer: ```suggestion
###### Playstyle Type
```

| Value | Name             | Title              | Subtitle                                                 | Emoji |
| ----- | ---------------- | ------------------ | -------------------------------------------------------- | ----- |
| 0     | NONE             |                    |                                                          |       |
| 1     | SOCIAL           | "Very Casual"      | "Skill doesn't matter. We have a good time win or lose." | 👋     |
| 2     | CASUAL           | "Casual"           | "We play to chill and have fun, but more fun if we win!" | 🎮     |
| 3     | COMPETITIVE      | "Competitive"      | "We play together to win, rank up, or beat challenges."  | ⚔️     |
| 4     | CREATIVE         |                    |                                                          |       |
| 5     | VERY_COMPETITIVE | "Very Competitive" | "We won't sleep until we win."                           | 💀     |

> ###### Badge Kind

Reviewer: ```suggestion
###### Badge Type
```

Reviewer: Wondering if these need descriptions, not sure tho

| Value | Name       |
| ----- | ---------- |
| 0     | SWORD      |
| 1     | WATER_DROP |
| 2     | SKULL      |
| 3     | TOADSTOOL  |
| 4     | MOON       |
| 5     | LIGHTNING  |
| 6     | LEAF       |
| 7     | HEART      |
| 8     | FIRE       |
| 9     | COMPASS    |
| 10    | CROSSHAIRS |
| 11    | FLOWER     |
| 12    | FORCE      |
| 13    | GEM        |
| 14    | LAVA       |
| 15    | PSYCHIC    |
| 16    | SMOKE      |
| 17    | SNOW       |
| 18    | SOUND      |
| 19    | SUN        |
| 20    | WIND       |

> ###### Banner Style

Reviewer: Same with these

| Value | Name        |
| ----- | ----------- |
| 0     | NIGHT_SKY   |
| 1     | CASTLE      |
| 2     | WORLD_MAP   |
| 3     | SEA_FOAM    |
| 4     | WARP_TUNNEL |
| 5     | HOUSE       |
| 6     | HEIGHT_MAP  |
| 7     | MESH        |
| 8     | SPATTER     |

## Endpoints

<RouteHeader method="POST" url="/clan/{guild.id}">

>   Convert guild to clan

Reviewer: ```suggestion
  Create Clan
```

</RouteHeader>

> Enables clan features for a guild

Reviewer: ```suggestion
Enables clan features for a guild. Returns a 204 empty response on success. Fires a [Guild Update](/topics/gateway-events#guild-update) Gateway event.
```
The response type is a guess based off the fact that the client ignores the response

| badge_color_primary   | string                                                                      | The primary color of the badge as a hexadecimal color string                |
| badge_color_secondary | string                                                                      | The secondary color of the badge as a hexadecimal color string              |
| banner                | integer                                                                     | Number representing a [banner style](#banner-style)                         |
| brand_color_primary   | string                                                                      | The primary color of the banner and some text as a hexadecimal color string |
| brand_color_secondary | string                                                                      | The secondary color of the banner as a hexadecimal color string             |
| badge_image           | [image data](/reference#cdn-data)                                           | The clan badge's image finalized as a PNG                                   |

<RouteHeader method="POST" url="/clan/{guild.id}/disable">
  Disable Clan
</RouteHeader>

> Disables a guild's clan features

Reviewer: ```suggestion
Disables a guild's clan features. Returns a 204 empty response on success. Fires a [Guild Update](/topics/gateway-events#guild-update) Gateway event.
```
